### PR TITLE
Add server code generation command

### PR DESCRIPTION
**Pull request by nati**
- added new command gohan generate to generate server side code
- added helper function for template command
- fixed required property bug on extending abstract schema
- added migraion create command by schema and doc

### CODE REVIEW

File: docs/cli.md [hunk 1]
```diff
@@ -363,4 +363,142 @@ Gohan will read this information and update the database accordingly.
...
+
+Setup mysql
+
+### Step2
+
+Prepare gohan.yaml with MySQL configuraion
+
+## Step3
+
+Prepare sqlboilar
```
Comment (marcin-ptaszynski) [inline docs/cli.md]: typo

File: cli/cli.go [hunk 2]
```diff
@@ -369,6 +375,85 @@ func getMigrateCommand() cli.Command {
 	}
 }
 
+func getCreateInitialMigrationCommand() cli.Command {
+	return cli.Command{
+		Name:        "create",
```
Comment (marcin-ptaszynski) [inline cli/cli.go]: Does `gohan migrate create` always generate initial migration? I think, it should be called `gohan migrate initial`, it leaves some space for diff subcommand. Goose file with some alter tables is a good start.

File: docs/cli.md [hunk 3]
```diff
@@ -363,4 +363,142 @@ Gohan will read this information and update the database accordingly.
...
+
+This command requires mysql setup, and generated code would work with only MySQL.
+
+### Step1
+
+Setup mysql
+
+### Step2
+
+Prepare gohan.yaml with MySQL configuraion
```
Comment (marcin-ptaszynski) [inline docs/cli.md]: typo

File: docs/cli.md [hunk 4]
```diff
@@ -363,4 +363,142 @@ Gohan will read this information and update the database accordingly.
...
+
+## Generate
+
+gohan generate command generates server side code.
+The idea is to genearate code which is fastest as possible.
+
+- Use echo framework which is fastest web framework in golang in early 2017
+- Trying to minimize use of interface
+- Use sqlboilar OR mapping framework
+[SQLBoilar](https://github.com/vattle/sqlboiler) sqlboilar framework generetes
```
Comment (marcin-ptaszynski) [inline docs/cli.md]: typo

File: docs/cli.md (hunk 1 above)
Comment (nati) [inline docs/cli.md]: done

File: docs/cli.md (hunk 3 above)
Comment (nati) [inline docs/cli.md]: done

File: docs/cli.md (hunk 4 above)
Comment (nati) [inline docs/cli.md]: done

File: cli/cli.go (hunk 2 above)
Comment (nati) [inline cli/cli.go]: done

**Pull request merged**